### PR TITLE
fix: Allow wrapper to work on production in localhost mode

### PR DESCRIPTION
**Pull request by pwnsdx**

### CODE REVIEW

**Comment by bennyn** [comment]
@Yserz Links are using the `?env=prod` parameter. Will this work with our future environment strategy?

**Comment by Yserz** [comment]
@bennyn the `env` parameter will be ignored. In order to have it working correctly in a development environment, one has to configure the `.env` file of the webapp project to use the correct environment.

File: electron/src/js/util.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ import * as url from 'url';
 
 import {pointInRectangle} from '../lib/pointInRect';
 
-const capitalize = (input: string): string => input.charAt(0).toUpperCase() + input.substr(1);
+const capitalize = (input: string): string => input.toLowerCase().replace(/\b[a-z]/g, letter => letter.toUpperCase());
```
Comment (ffflorian) [inline electron/src/js/util.ts]: What's the benefit of that?

File: electron/src/menu/developer.ts [hunk 2]
```diff
@@ -64,16 +63,20 @@ const devToolsTemplate: Electron.MenuItemConstructorOptions = {
...
-  return {
-    checked: currentEnvironment === env,
-    click: () => {
-      environment.setEnvironment(env);
-      getPrimaryWindow().reload();
-    },
-    label: util.capitalize(env),
-    type: 'radio',
-  };
+const createEnvironmentsTemplate = () => {
```
Comment (lipis) [inline electron/src/menu/developer.ts]:
```suggestion
const createEnvironmentTemplates = () => {
```

File: electron/src/menu/developer.ts [hunk 3]
```diff
@@ -99,19 +102,14 @@ const menuTemplate: Electron.MenuItemConstructorOptions = {
     devToolsTemplate,
     reloadTemplate,
     separatorTemplate,
-    createEnvironmentTemplate(environment.TYPE.PRODUCTION),
-    createEnvironmentTemplate(environment.TYPE.INTERNAL),
-    createEnvironmentTemplate(environment.TYPE.STAGING),
-    createEnvironmentTemplate(environment.TYPE.DEV),
-    createEnvironmentTemplate(environment.TYPE.EDGE),
-    createEnvironmentTemplate(environment.TYPE.LOCALHOST),
+    ...createEnvironmentsTemplate(),
```
Comment (lipis) [inline electron/src/menu/developer.ts]:
```suggestion
    ... createEnvironmentTemplates(),
```

Comment (lipis) [inline electron/src/menu/developer.ts]:
```suggestion
    ...createEnvironmentTemplates(),
```

File: electron/src/main.ts [hunk 4]
```diff
@@ -221,23 +221,26 @@ const showMainWindow = (mainWindowState: WindowStateKeeper.State) => {
...
-  main.webContents.session.webRequest.onHeadersReceived(
-    {
-      urls: ['https://staging-nginz-https.zinfra.io/*'],
-    },
-    (details: OnHeadersReceivedDetails, callback: OnHeadersReceivedCallback) => {
-      if (environment.getEnvironment() === environment.TYPE.LOCALHOST) {
+  if (
+    environment.getEnvironment() == environment.TYPE.LOCALHOST ||
+    environment.getEnvironment() == environment.TYPE.LOCALHOST_PRODUCTION
+  ) {
```
Comment (ffflorian) [inline electron/src/main.ts]: Let's create a variable for that before the `if` case.

**Pull request merged**